rskj-core/src/main/java/co/rsk/net/sync/PeersInformation.java

```diff
@@ -106,6 +109,25 @@ public SyncPeerStatus getPeer(Peer peer) {
     }
 
     public Optional<Peer> getBestPeer() {
+        if (percentageOfPeersToConsiderInRandomSelection > 0.0D) {
```

Comment:
could you please cover edge cases of this part with unit tests?

Reply:
I will add unit tests when we agree on the implementation